src/data_provider/Extractor.py

```diff
@@ -7,7 +9,18 @@ def __init__(self):
         pass
 
     @staticmethod
-    def get_day_to_services_dict(calendar_df: pd.DataFrame):
+    def get_adjacent_stops_dict(stops_df: pd.DataFrame) -> dict:
+        adjacent_stops = {}
+        for id_1, _, lat_1, lon_1, _, _ in stops_df.itertuples():
+            for id_2, _, lat_2, lon_2, _, _ in stops_df.itertuples():
+                duration = Extractor._get_walking_time(lon_1, lat_1, lon_2, lat_2)
```

Comment:
this line of code runs twice for each stops pair

Reply:
I know, but we neeed it for directed edges,